### PR TITLE
Fix HasGetInfo for new bitcoind wallets.

### PR DESCRIPTION
**Pull request by Tanariel**
getInfo has been split into 3 new calls for the last version of bitcoind, this patch change the behavior or wallet-rpc whenthe coin don't have getInfo.

### CODE REVIEW

File: web/yaamp/core/rpc/wallet-rpc.php [hunk 1]
```diff
@@ -377,7 +379,35 @@ function __call($method, $params)
...
-		$res = $this->rpc->__call($method,$params);
+        switch ($method) {
+            case 'getinfo':
+                if ($this->hasGetInfo) {
+                    $res = $this->rpc->__call($method,$params);
+                } else {
+                    $miningInfo = $this->rpc->getmininginfo();
+                    $res["blocks"] = arraySafeVal($miningInfo,"blocks");
+                    $res["difficulty"] = arraySafeVal($miningInfo,"difficulty");
+                    $res["testnet"] = arraySafeVal($miningInfo,"chain") == "main" ? 'true' : 'false';
```
[the remark this answers is not in the document]
Comment (Tanariel) [reply web/yaamp/core/rpc/wallet-rpc.php]: Damn ! :)

**Comment by Tanariel** [comment]
Change done as requested.

File: web/yaamp/core/rpc/wallet-rpc.php [hunk 2]
```diff
@@ -377,7 +379,34 @@ function __call($method, $params)
...
-		$res = $this->rpc->__call($method,$params);
+        	switch ($method) {
+			case 'getinfo':
+				if ($this->hasGetInfo) {
+					$res = $this->rpc->__call($method,$params);
+				} else {
+					$miningInfo = $this->rpc->getmininginfo();
+					$res["blocks"] = arraySafeVal($miningInfo,"blocks");
+					$res["difficulty"] = arraySafeVal($miningInfo,"difficulty");
+					$res["testnet"] = "main" == arraySafeVal($miningInfo,"chain") ? 'false' : 'true';
```
Comment (tpruvot) [inline web/yaamp/core/rpc/wallet-rpc.php]: true and false are ok in php... not as string

Comment (tpruvot) [reply web/yaamp/core/rpc/wallet-rpc.php]: `$res["testnet"] = arraySafeVal($miningInfo,"chain") != "main";`

**Comment by Tanariel** [comment]
Modified as requested :)

**Comment by tpruvot** [comment]
tx for your patience ;) merging that

**Comment by sp3sh** [comment]
This fix does not seem to work on DGB - ERROR error -32601: getinfo this call was removed in version 6.16.0. use the appropriate fields from: - getblockchaininfo: blocks, difficulty, chain - getnetworkinfo: version, protocolversion, timeoffset, connections, proxy, relayfee, warnings - getwalletinfo: balance, keypoololdest, keypoolsize, paytxfee, unlocked_until, walletversion digibyte-cli has the option -getinfo to collect and format these in the old format.
image

**Pull request merged**